### DISCUSSION TITLE
PhantomCamera3D limits?

### DISCUSSION THREAD

**Discussion by Mangonels** [ramokz/phantom-camera#345] [Question / Help Wanted] [answered]
What's the approach for limiting the movement within a collision shape for a 3D phantom camera? If any.

**Comment by ramokz** [comment, marked as answer]
It's not currently possible in 3D scenes, but have thought about making a system to support it.

**Reply by ramokz** [reply]
Though, you could forcefully clamp the position of a given `PCam3D` manually via an external script. Obviously, not overly user-friendly or easy to set up.

**Reply by natinusala** [reply]
Could you share more details about how this can be done? I have a camera angled 90° down so I only have to clamp X and Z axis. I'm using follow mode.

If I add a script to a node in my scene that sets the position of the `Camera3D` node, it only works if the character is not moving. As soon as I start moving, the phantom camera movements takes precedence over my clamp. When I release the controller stick the character stops and the camera snaps back to my clamped position. If I try to set the position of the phantom camera directl, it doesn't do anything.

Thanks!

**Reply by ramokz** [reply]
I remember having a simple solution for this a while back, but can't seem to replicate anymore when trying again.
So there might have been some changes to the codebase that made that a non-trivial thing to DIY.

In any case, I do want to support this officially, as there have been other discussions (outside of this post) that have mentioned it as well, and think there's value in having it.

**Comment by KyleWelfare** [comment]
+1 for this idea, would be very helpful 😄
